third_party/ruby/ruby_3_3.BUILD

```diff
@@ -0,0 +1,72 @@
+# vim: ft=bzl sw=4 ts=4 et
+
+load("@com_stripe_ruby_typer//third_party/ruby:build-ruby.bzl", "ruby")
+
+ruby(
+    append_srcs = [],
+    configure_flags = [
+        "--enable-shared",
+        "--sysconfdir=/etc",
+        "--localstatedir=/var",
+        "--disable-maintainer-mode",
+        "--disable-dependency-tracking",
+        "--disable-install-doc",
+        "--without-git",
+    ] + select({
+        # Enforce that we don't need Ruby to build in release builds.
+        # (In non-release builds, we allow for an available system Ruby to
+        # speed up the build.)
+        "@com_stripe_ruby_typer//tools/config:release": ["--with-baseruby=no"],
+        "//conditions:default": [],
+    }) + select({
+        # Do not enable the JIT unless opted in.
+        "@com_stripe_ruby_typer//tools/config:jit_enabled": ["--enable-yjit=stats"],
+        "//conditions:default": ["--disable-jit-support"],
+    }),
+    copts = [
+        "-g",
+        "-fstack-protector-strong",
+        "-Wformat",
+        "-Werror=format-security",
+        # The Ruby build is very noisy without this flag.
+        "-Wno-compound-token-split-by-macro",
+    ] + select({
+        "@com_stripe_ruby_typer//tools/config:dbg": [],
+        "//conditions:default": ["-O2"],
+    }),
+    cppopts = [
+        "-Wdate-time",
+        "-D_FORTIFY_SOURCE=2",
+        "-fPIC",
+    ] + select({
+        # Don't include JIT statistics unless we're building JIT support
+        "@com_stripe_ruby_typer//tools/config:jit_enabled": ["-DYJIT_STATS=1"],
+        "//conditions:default": [],
```

Comment:
OOC, do you need both `--enable-yjit=stats` and `-DYJIT_STATS=1`?  Seems a little weird, but OK.

Is having stats enabled going to mess with any performance numbers that you get?